### ISSUE TITLE
Ability to set the destination filename

### ISSUE THREAD

**Issue by zkochan** [schnittstabil/lnk#1]
I'd like to use this package in [pnpm](https://github.com/rstacruz/pnpm), but I need the possibility to set the symlink filename. Currently it takes the source filename and that is not good for all of my use cases.

So instead of this:

```js
lnk('assets/favicon.ico', 'dist')
```

I'd need something like

```js
lnk('assets/favicon.ico', 'dist/newfavicon.ico')
```

I don't know what would be the best way to add this feature. Maybe changing the default behavior? That would be closer to the `fs.symlink` API

**Comment by schnittstabil** [comment]
Sadly, `'dist/newfavicon.ico'` would be ambiguous, it could denote a directory or a file path.

I would opt for a `rename` option (`string` &cup; `function`):

```js
// string
lnk('assets/favicon.ico', 'dist', {rename: 'subpath/newfavicon.ico'});
// => 'dist/subpath/newfavicon.ico'

// function
lnk(globby('assets/fav*.ico'), 'dist', {
    rename: ({dirname, basename}) => `${dirname}/new-${basename}`
    // or
    rename: ({path}) => …
});
```

Implementing the `function` feature would take some time, but the `string` feature would be simple. @zkochan Would this work for you?

**Comment by zkochan** [comment]
Actually, for now the existing API worked for me:
https://github.com/rstacruz/pnpm/blob/pure-store/src/install/symlinkToModules.ts#L46

For now I will use both this package and [link-dir](github.com/zkochan/link-dir). It has symlink overwriting, which we also need in pnpm.

> Sadly, 'dist/newfavicon.ico' would be ambiguous, it could denote a directory or a file path.

Unless there's a breaking change and the function always expects the full path, right? That would also mean that the first argument can't be an array of targets anymore.

I like the first solution with the string. Could it be a 3rd parameter maybe?

```js
lnk('assets/favicon.ico', 'dist', 'subpath/newfavicon.ico');
// => 'dist/subpath/newfavicon.ico'
```

**Comment by schnittstabil** [comment]
> That would also mean that the first argument can't be an array of targets anymore.

`lnk` allows arrays because this simplifies many use-cases, especially in conjunction with the [`parents` option](https://github.com/schnittstabil/lnk#parents).

> Could it be a 3rd parameter maybe?

IMO, that would be too cumbersome; and moreover, it could be confused with the `cwd` option as well.

---

> For now I will use both this package and link-dir. It has symlink overwriting, which we also need in pnpm.

It is not clear to me, how `lnk` and `link-dir` differ. `lnk` offers a [`force` option](https://github.com/schnittstabil/lnk#force) for overwriting too.

From the source you've linked:
```ts
return linkDir(absolutePath, dest)
```

At a first glance this could also be done with `lnk` by:
```ts
return lnk(absolutePath, dest, {type: 'junction'})
```

**Comment by zkochan** [comment]
ok, seems like `opts.rename` is the best solution in this situation.

> It is not clear to me, how lnk and link-dir differ. lnk offers a force option for overwriting too.

Thanks, I'll check the force option today, maybe that will do

**Comment by schnittstabil** [comment]
Landed with `v1.1.0` – see [`opts.rename`](https://github.com/schnittstabil/lnk#rename) for details.

@zkochan I hope these work for you:

```js
lnk('assets/favicon.ico', 'dist', {rename: 'newfavicon.ico'});

// or e.g. with globs
lnk(globby('assets/fav*.ico'), 'dist', {
    rename: ({base}) => `new${base}`
});
```

**Comment by zkochan** [comment]
Thank you!
